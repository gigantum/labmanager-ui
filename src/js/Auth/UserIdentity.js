//vendor
import {
  graphql,
} from 'react-relay'
//mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//environment
import {fetchQuery} from 'JS/createRelayEnvironment';

const userIdentityQuery = graphql`
  query UserIdentityQuery{
    userIdentity{
      id
      username
      email
      givenName
      familyName
      isSessionValid
    }
  }
`;

const UserIdentity = {
  getUserIdentity: () => {

    return new Promise((resolve, reject) =>{

      let fetchData = function(){

        fetchQuery(userIdentityQuery(), {}).then((response, error) => {

          if(response){
            resolve(response)
          }else{

            reject(response)
          }
        }).catch((error) =>{

          reject(error)
        })
      }

      fetchData()
    })
  },
  removeUserIdentity: () => {
    RemoveUserIdentityMutation(()=>{
      console.log('removed user')
    })
  }

}

export default UserIdentity
