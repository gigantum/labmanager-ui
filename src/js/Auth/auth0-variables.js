export const AUTH_CONFIG = {
  domain: 'calum.auth0.com',
  clientId: '9wFN5CZA4Z9xa1r8KZIZbevpjHG7bAab',
  callbackUrl: window.location.origin + '/callback'
}
