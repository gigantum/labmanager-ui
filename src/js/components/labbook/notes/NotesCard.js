import React, { Component } from 'react'
import dateformat from 'dateformat'


export default class NotesCard extends React.Component {
  constructor(props){
  	super(props);
    this.state = {showExtraInfo: false}
  }
  /*
    function(): reverse state of showExtraInfo
  */
  _toggleExtraInfo(){
    this.setState({showExtraInfo: !this.state.showExtraInfo})
  }
  /*
    function(string): inputs a timestamp
    return (string) returns a string in the for of 'X {seconds, hours, days, months, years} ago'
  */
  _timeAgo(time){
    let units = [
      { name: "second", limit: 60, in_seconds: 1 },
      { name: "minute", limit: 3600, in_seconds: 60 },
      { name: "hour", limit: 86400, in_seconds: 3600  },
      { name: "day", limit: 604800, in_seconds: 86400 },
      { name: "week", limit: 2629743, in_seconds: 604800  },
      { name: "month", limit: 31556926, in_seconds: 2629743 },
      { name: "year", limit: null, in_seconds: 31556926 }
    ];
    let diff = (new Date() - new Date(time)) / 1000;
    if (diff < 5) return "now";

    let i = 0;
    let unit;
    while(unit = units[i++]) {
      if (diff < unit.limit || !unit.limit){
        let displayTime =  Math.floor(diff / unit.in_seconds);
        return displayTime + " " + unit.name + (diff>1 ? "s" : "") + " ago";
      }
    }
  }

  _getTimeOfDay(timestamp){
    let time = new Date(timestamp);
    return (time.getHours()%12) + ':' + ((time.getMinutes() > 9) ? time.getMinutes() : '0' + time.getMinutes()) + (time.getHours() > 12 ? 'pm' : 'am');
  }

  render(){
    return(
        <div className="NotesCard card">

          <div className="flex flex--row justify--space-between">
              <p className="NotesCard__time">
                {this._getTimeOfDay(this.props.edge.node.timestamp)}
              </p>
              <p className="NotesCard__commit-message">
                {this.props.edge.node.message}
              </p>

              <div className={!this.state.showExtraInfo ? "NotesCard__toggle-button closed flex justify--space-around": "NotesCard__toggle-button open flex justify--space-around"}
                  onClick={() => this._toggleExtraInfo()}
              >
              Activity Log
              <div className="NotesCard__toggle-icon"></div>
              </div>
          </div>

          <div className={this.state.showExtraInfo ? 'NotesCard__expanded-view' : 'NotesCard__expanded-view no-height'}>
            <p>
              Commit Id: {this.props.edge.node.commit}
            </p>
            <p>
              Level: {this.props.edge.node.level}
            </p>
            <p>
              {this.props.edge.node.linkedCommit}
            </p>
            <p>
              {dateformat(this.props.edge.node.timestamp, "dddd, mmmm dS, yyyy, h:MM:ss TT")}
            </p>

            <div>
              <ul className="NotesCard__tags-list flex flex--row flex--wrap">
                <div>Tags: {' '}</div>
                {this.props.edge.node.tags.map((tag, index) => {
                  return(
                    <li
                      key={tag + index + this.props.edge.node.commit}
                    >
                      {tag}
                    </li>)
                  })
                }
              </ul>
            </div>

          </div>

      </div>
    )
  }
}
