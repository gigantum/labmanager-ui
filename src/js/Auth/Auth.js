import history from 'JS/history';
import auth0 from 'auth0-js';
import { AUTH_CONFIG } from './auth0-variables';
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation'
//store
import store from 'JS/redux/store'


export default class Auth {
  auth0 = new auth0.WebAuth({
    domain: AUTH_CONFIG.domain,
    clientID: AUTH_CONFIG.clientId,
    redirectUri: AUTH_CONFIG.callbackUrl,
    audience:  AUTH_CONFIG.audience,
    responseType: 'token id_token',
    scope: 'openid profile email user_metadata'
  });

  constructor() {
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.handleAuthentication = this.handleAuthentication.bind(this);
    this.isAuthenticated = this.isAuthenticated.bind(this);

  }

  login() {
    this.auth0.authorize();
  }

  handleAuthentication() {

    this.auth0.parseHash((err, authResult) => {
      if (authResult && authResult.accessToken && authResult.idToken) {

        this.setSession(authResult);

      } else if (err) {
        console.error(err);
      //  alert(`Error: ${err.error}. Check the console for further details.`);
      }
    });
  }

  setSession(authResult) {

    // Set the time that the access token will expire at
    let expiresAt = JSON.stringify((authResult.expiresIn * 1000) + new Date().getTime());
    localStorage.setItem('access_token', authResult.accessToken);
    localStorage.setItem('id_token', authResult.idToken);
    localStorage.setItem('expires_at', expiresAt);
    localStorage.setItem('family_name', authResult.idTokenPayload.family_name);
    localStorage.setItem('given_name', authResult.idTokenPayload.given_name);
    localStorage.setItem('email', authResult.idTokenPayload.email);
    localStorage.setItem('username', authResult.idTokenPayload.nickname);
    //redirect to labbooks when user logs in

    let storrageRoute = sessionStorage.getItem('CALLBACK_ROUTE')
    let route = storrageRoute !== '' ? storrageRoute : `/labbooks`

    history.replace(route)
  }

  logout() {

    localStorage.removeItem('access_token');
    localStorage.removeItem('id_token');
    localStorage.removeItem('expires_at');
    localStorage.removeItem('family_name')
    localStorage.removeItem('given_name');
    localStorage.removeItem('email');
    localStorage.removeItem('username');
    RemoveUserIdentityMutation(()=>{
      //redirect to root when user logs out
      history.replace('/');
    })
  }

  isAuthenticated() {
    // Check whether the current time is past the
    // access token's expiry time
    let expiresAt = JSON.parse(localStorage.getItem('expires_at'));

    return new Date().getTime() < expiresAt;
  }
}
