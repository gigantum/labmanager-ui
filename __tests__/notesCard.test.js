import React from 'react';
import NotesCard from './../src/js/components/labbook/notes/NotesCard';
import renderer from 'react-test-renderer';
import Auth from './../src/js/Auth/Auth';

let edge ={
  node: {
    "linkedCommit": "dbbc8891ada166a1da21d874e02c8f5085ba514c",
    "commit": "acad13eee2a5959bdd594aa89494564b60604880",
    "level": "USER_MINOR",
    "tags": [
      "my tag",
      "other tag"
    ],
    "timestamp": "2017-07-26T11:56:00-04:00",
    "message": "User Returned a result",
    "id": "Tm90ZTpkZWZhdWx0JmxhYm9vazQmYWNhZDEzZWVlMmE1OTU5YmRkNTk0YWE4OTQ5NDU2NGI2MDYwNDg4MA==",
    "author": "noreply@gigantum.io"
  },
  "cursor": "MQ=="
}

test('Test note card for rendering', () => {

  //const isAuthenticated = function(){return true};
  const component = renderer.create(
    <NotesCard edge={edge} key={"id"}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
