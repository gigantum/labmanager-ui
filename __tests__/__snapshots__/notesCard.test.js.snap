// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test note card for rendering 1`] = `
<div
  className="NotesCard card"
>
  <div
    className="flex flex--row justify--space-between"
  >
    <div
      className="flex flex--column justify--space-between"
    >
      <p>
        User Returned a result
      </p>
      <p>
        1 weeks ago
      </p>
    </div>
    <div
      className="flex flex--column justify--space-between"
    >
      <p>
        Commit Id: 
        acad13eee2a5959bdd594aa89494564b60604880
      </p>
      <div
        className="NotesCard__toggle-button"
        onClick={[Function]}
        style={
          Object {
            "background": "url(down-arrow.svg) no-repeat",
          }
        }
      />
    </div>
  </div>
  <div
    className="NotesCard__expanded-view no-height"
  >
    <p>
      Level: 
      USER_MINOR
    </p>
    <p
      className="USER_MINOR"
    >
      Commit Id: 
      Tm90ZTpkZWZhdWx0JmxhYm9vazQmYWNhZDEzZWVlMmE1OTU5YmRkNTk0YWE4OTQ5NDU2NGI2MDYwNDg4MA==
    </p>
    <p>
      dbbc8891ada166a1da21d874e02c8f5085ba514c
    </p>
    <p>
      Wednesday, July 26th, 2017, 11:56:00 AM
    </p>
    <div>
      <ul
        className="NotesCard__tags-list flex flex--row flex--wrap"
      >
        <div>
          Tags: 
           
        </div>
        <li>
          my tag
        </li>
        <li>
          other tag
        </li>
      </ul>
    </div>
  </div>
</div>
`;
